backend/src/main/java/de/jankahle/capstone/service/RecipeService.java

```diff
@@ -31,7 +31,9 @@ public RecipeService(RecipeMongoDB recipeMongoDB, ImageReaderService imageReader
 
     public Recipe saveRecipe(RecipeDto recipeDto) {
         Recipe recipe = recipeDto.toRecipe();
-        recipe.setId(IdUtility.generateId());
+        if (recipe.getId().isEmpty()) {
```

Comment:
You have a `Recipe`,  `RecipeDto` and a `DBRecipe` which, from a naming perspective, seems strange. Maybe rename to `RecipeDao` ?

Reply:
Good Point. This was just me being uncreative.

Reply:
What happens in the zone stays in the zone!